rpc-server/src/modules/blocks/utils.rs

```diff
@@ -2,10 +2,54 @@ use crate::config::ServerContext;
 use crate::modules::blocks::methods::fetch_block;
 use crate::modules::blocks::CacheBlock;
 use crate::storage::ScyllaDBManager;
-use anyhow::Context;
 use near_primitives::views::{StateChangeValueView, StateChangesRequestView};
 use num_traits::ToPrimitive;
 
+/// Fetch object from s3 with retry
+/// Try to get 3 times if we have some problems with network
+async fn fetch_object_from_s3_with_retry<T>(
```

Comment:
note for the future: it's much better to open separate PRs for separate issues. This makes the merging faster (faster dopamine + 2 doses of it because we will merge 2 PRs instead of 1)
And it's much easier for the reviewers to look at smaller PRs

(I'm not suggesting splitting this exact PR, let's finish it as it is)